src/metadata.rs

```diff
@@ -61,6 +62,43 @@ impl Value {
     }
 }
 
+macro_rules! from_value {
+    (($name:ident) { $($typ:ty => $ex:expr),+ $(,)? }) => {
+        $(
+            impl From<$typ> for Value {
+                fn from($name: $typ) -> Self {
+                    $ex
+                }
+            }
+        )+
+    }
+}
+
+from_value! {
+    (value) {
+        bool => Self::Bool(value),
+        u64 => Self::Number(value),
+        Vec<Self> => Self::List(value),
+        String => Self::String(value),
+        &str => Self::String(value.into()),
+        bytes::Bytes => Self::Bytes(value),
+    }
+}
+
+impl PartialEq for Value {
+    fn eq(&self, other: &Self) -> bool {
+        match (self, other) {
+            (Self::Bool(a), Self::Bool(b)) => a == b,
+            (Self::Number(a), Self::Number(b)) => a == b,
+            (Self::List(a), Self::List(b)) => a == b,
+            (Self::String(a), Self::String(b)) => a == b,
+            (Self::Bytes(a), Self::Bytes(b)) => a == b,
+            (Self::String(a), Self::Bytes(b)) | (Self::Bytes(b), Self::String(a)) => a == b,
+            _ => false,
```

Comment:
> its not too far from doing vstring.as_bytes() == vbytes or similar wdyt?

Well it is more involved than that, because we'd have to check the variants as well whenever we wanted to do that check.

I think I'd rather just remove the `_ => false` branch and write out the rest of the variants, so we don't introduce a bug, because that is a good point.

Reply:
> I think I'd rather just remove the _ => false branch and write out the rest of the variants, so we don't introduce a bug, because that is a good point.

👍🏻 I like it. Then if/when new variants show up, they have to be explictly handled.

Reply:
Listing them explicitly SGTM!